core/src/main/java/com/github/mizool/core/exception/RuleViolationException.java

```diff
@@ -0,0 +1,49 @@
+/**
+ *  Copyright 2017 incub8 Software Labs GmbH
+ *  Copyright 2017 protel Hotelsoftware GmbH
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.github.mizool.core.exception;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class RuleViolationException extends RuntimeException
+{
+    private final Set<RuleViolation> ruleViolations;
+
+    public RuleViolationException(String message, Set<RuleViolation> ruleViolations)
+    {
+        super(message);
+
+        if (ruleViolations == null)
+        {
+            this.ruleViolations = null;
+        }
+        else
+        {
+            this.ruleViolations = new HashSet<>(ruleViolations);
```

Comment:
this would allow manipulating the set as `getRuleViolations()` returns it without protecting it. make an `ImmutableSet` copy instead of using `HashSet`.